### ISSUE TITLE
Websocket closes connection despite heartbeat when client runs long-running (but asynchronous) task

### ISSUE THREAD

**Issue by PhracturedBlue** [fafhrd91/aiohttp#7508]
### Describe the bug

When using websockets and both client and server have hearbeat enablesd, the socket is closed when the client runs a long-running (but asynchronous) task

### To Reproduce

Client/server code:
```
import sys
import time
import asyncio
import aiohttp
import aiohttp.web
import logging

PORT = 9090
logging.basicConfig(level=logging.INFO, format="%(asctime)s.%(msecs)d %(levelname)s: %(message)s", datefmt="%Y-%m-%d %H:%M:%S")

async def websocket_handler(request):
    logging.info("connected")
    ws_ = aiohttp.web.WebSocketResponse(autoping=True, heartbeat=60)
    await ws_.prepare(request)
    logging.info("ws established")
    async for msg in ws_:
        logging.info(f"Got message: {msg}")
        await asyncio.sleep(15)
        await ws_.send_json({'resp': 1})
        logging.info(f"Sent response")
    logging.info(f"closed: {ws_.exception()}")

async def server():
    app = aiohttp.web.Application()
    app.add_routes([aiohttp.web.get('/1', websocket_handler)])
    runner = aiohttp.web.AppRunner(app)
    await runner.setup()
    site = aiohttp.web.TCPSite(runner, "0.0.0.0", PORT)
    await site.start()
    logging.info(f"Listening on {PORT}")
    while True:
        await asyncio.sleep(1)

async def client():
    async with aiohttp.ClientSession() as session:
        async with session.ws_connect(f"http://localhost:{PORT}/1", heartbeat=60) as ws_:
            logging.info("connected")
            await asyncio.sleep(15)
            await ws_.send_json({'key': 'value'})
            await ws_.receive()
            logging.info("got response")
            await asyncio.sleep(300)
    logging.info("Exiting")

if len(sys.argv) > 1:
    asyncio.run(server())
else:
    asyncio.run(client())
```
Start server via: `python3 bug.py server`
Start client via: `python3 bug.py`

Server output:
```
2023-08-10 18:05:15.242 INFO: Listening on 9090
2023-08-10 18:05:24.841 INFO: connected
2023-08-10 18:05:24.842 INFO: ws established
2023-08-10 18:05:39.858 INFO: Got message: WSMessage(type=<WSMsgType.TEXT: 1>, data='{"key": "value"}', extra='')
2023-08-10 18:05:54.859 INFO: Sent response
2023-08-10 18:07:26.173 INFO: closed: Cannot write to closing transport
```

Client output:
```
2023-08-10 18:05:24.842 INFO: connected
2023-08-10 18:05:54.860 INFO: got response
```

### Expected behavior

heartbeat should continue to function while long-running async task runs.  Socket should not be closed

### Logs/tracebacks

```python-traceback
None
```

### Python Version

```console
$ python --version
Python 3.10.4
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.8.4
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author:
Author-email:
License: Apache 2
Location: /work/ghaushe/git/lumberyard_1/packages/lib/python3.10/site-packages
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by: pytest-aiohttp
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 6.0.4
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /work/ghaushe/git/lumberyard_1/packages/lib/python3.10/site-packages
Requires:
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.9.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /work/ghaushe/git/lumberyard_1/packages/lib/python3.10/site-packages
Requires: idna, multidict
Required-by: aiohttp
```

### OS

Centos7.9 on x86_64

### Related component

Server, Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by PhracturedBlue** [comment]
I assume this is probably user-error.  Looking at the code, the pong() is only processed by autoping in receive() which is only called by the iterator, which is obvioulsy not called if a long-running task is executing.

It is unexpected (to me) behavior though, as the heartbeat seems to be sent during long-running tasks, but the response requires a call to receive().  Should this be better documented?

**Comment by Dreamsorcerer** [comment]
Duplicate of https://github.com/aio-libs/aiohttp/discussions/7320#discussioncomment-6144209
If you'd like to contribute some improvements to the docs, so other users don't get confused, that'd be really useful.
